instance/tests/models/test_openedx_instance.py

```diff
@@ -1251,4 +1253,33 @@ def test_active_vm_dns_records(self, mocks, mock_consul):
         self._verify_vm_dns_records('test.com', [appserver1.server.public_ip])
         appserver2.make_active()
         self._verify_vm_dns_records('test.com', [appserver1.server.public_ip, appserver2.server.public_ip])
-        # We haven't implemented cleaning up VM DNS records yet, so we can't test it.
+
+    @patch('instance.models.mixins.load_balanced.LoadBalancedInstance.remove_dns_records')
+    @patch('instance.models.mixins.openedx_monitoring.OpenEdXMonitoringMixin.disable_monitoring')
+    @patch('instance.models.load_balancer.LoadBalancingServer.reconfigure')
+    @patch_services
+    def test_clean_up_dns_records(self, mocks, *mock):
```

Comment:
@pkulkark, this test can be simplified to only check whether all the DNS records for the active appservers of an instance are removed during archival.

The checks to the active VM DNS records on various sequences of activation, deactivation, can be moved to the test above. Also, please test the `instance must have 2 active appservers at all the times` scenario in which there will 4 active VMs temporarily before the 2 older ones are deactivated.

Reply:
Modified it. Let me know if that's better?